### PR TITLE
feat: improvements

### PR DESCRIPTION
**Pull request by johnny243**
- update to latest component relay
- make optional delegate functions if not using Filesafe
- expose `platform` and `environment` from component relay

[Internal link](https://app.asana.com/0/1201611182420182/1201680328435447)

### CODE REVIEW

File: src/editorKit.ts [hunk 1]
```diff
@@ -187,7 +187,7 @@ export default class EditorKitBase {
       }
 
       if (isNewNoteLoad) {
-        this.delegate.clearUndoHistory()
+        this.delegate.clearUndoHistory && this.delegate.clearUndoHistory()
```
Comment (atmoio) [inline src/editorKit.ts]:
```suggestion
        this.delegate.clearUndoHistory?.()
```

File: src/editorKit.ts [hunk 2]
```diff
@@ -201,6 +201,22 @@ export default class EditorKitBase {
   }
 
   private configureFileSafe() {
+    const delegateFunctions = [
```
Comment (atmoio) [inline src/editorKit.ts]:
```suggestion
    const requiredDelegateFunctions = [
```

File: src/editorKit.ts [hunk 3]
```diff
@@ -201,6 +201,22 @@ export default class EditorKitBase {
...
+      'getCurrentLineText',
+      'getPreviousLineText',
+      'replaceText',
+      'getElementsBySelector',
+      'insertElement',
+      'preprocessElement',
+      'insertRawText'
+    ]
+
+    for (const theFunction of delegateFunctions) {
```
Comment (atmoio) [inline src/editorKit.ts]:
```suggestion
    for (const functionName of delegateFunctions) {
```

File: src/editorKit.ts [hunk 4]
```diff
@@ -23,22 +23,22 @@ import type { ItemMessagePayload } from '@standardnotes/snjs'
  * with the appropriate callbacks.
  */
 export interface EditorKitDelegate {
-  insertRawText: (text: string) => void
+  insertRawText?: (text: string) => void
```
Comment (atmoio) [inline src/editorKit.ts:26]: If this is required why make it optional?

Comment (johnny243) [reply src/editorKit.ts:26]: This is required when `supportsFileSafe` is `true` only.

Comment (atmoio) [reply src/editorKit.ts:26]: Ah ok

Comment (atmoio) [reply src/editorKit.ts:26]: You could possibly make a separate interface then for FileSafe functions

**Review by atmoio — APPROVED** [review]

File: src/editorKit.ts (hunk 4 above)
Comment (johnny243) [reply src/editorKit.ts:26]: I wonder if we could use conditional types for this. Example, when `supportsFileSafe` is `true`, we should ask for the FileSafe functions. Will take a deeper look at this later.

**Pull request merged**